src/main/java/uk/gov/hmcts/darts/audio/helper/UnstructuredDataHelper.java

```diff
@@ -0,0 +1,146 @@
+                eodEntity.getMedia().getId(),
+                uuid
+            );
+        } catch (Exception e) {
+            log.error(
+                "Upload to unstructured data store failed for mediaRequestId {}.",
+                eodEntity.getMedia().getId()
+            );
+        }
+        return uuid;
+    }
+
+    @Transactional
+    private void saveToDatabase(ExternalObjectDirectoryEntity eod, UUID uuid) {
+        ExternalObjectDirectoryEntity eodUnstructured = new ExternalObjectDirectoryEntity();
+
+        MediaEntity mediaEntity = eod.getMedia();
+        if (mediaEntity != null) {
+            eodUnstructured.setMedia(mediaEntity);
+        }
+        TranscriptionDocumentEntity transcriptionDocumentEntity = eod.getTranscriptionDocumentEntity();
+        if (transcriptionDocumentEntity != null) {
+            eodUnstructured.setTranscriptionDocumentEntity(transcriptionDocumentEntity);
+        }
+        AnnotationDocumentEntity annotationDocumentEntity = eod.getAnnotationDocumentEntity();
+        if (annotationDocumentEntity != null) {
+            eodUnstructured.setAnnotationDocumentEntity(annotationDocumentEntity);
+        }
+        CaseDocumentEntity caseDocumentEntity = eod.getCaseDocument();
+        if (caseDocumentEntity != null) {
+            eodUnstructured.setCaseDocument(caseDocumentEntity);
+        }
+        eodUnstructured.setExternalLocation(uuid);
+        OffsetDateTime now = OffsetDateTime.now();
+        eodUnstructured.setCreatedDateTime(now);
+        eodUnstructured.setLastModifiedDateTime(now);
+        eodUnstructured.setChecksum(eod.getChecksum());
+        ObjectRecordStatusEntity storedStatus = objectRecordStatusRepository.getReferenceById(STORED.getId());
+        ExternalLocationTypeEntity unstructuredType = externalLocationTypeRepository.getReferenceById(UNSTRUCTURED.getId());
+        eodUnstructured.setStatus(storedStatus);
+        eodUnstructured.setExternalLocationType(unstructuredType);
+        eodUnstructured.setVerificationAttempts(1);
+        var systemUser = userAccountRepository.getReferenceById(SystemUsersEnum.DEFAULT.getId());
+        eodUnstructured.setCreatedBy(systemUser);
+        eodUnstructured.setLastModifiedBy(systemUser);
+        externalObjectDirectoryRepository.save(eodUnstructured);
+        log.debug(
+            "Created new record in EOD mediaRequestId {}. External location: {}",
+            eodUnstructured.getMedia().getId(),
+            eodUnstructured.getExternalLocation()
+        );
+    }
+
+    @Transactional
+    private void removeFromDatabase(ExternalObjectDirectoryEntity eodEntityToDelete) {
+        if (eodEntityToDelete != null) {
+            externalObjectDirectoryRepository.delete(eodEntityToDelete);
+            log.debug(
+                "Deleted old unstructured EOD mediaRequestId {}. External location: {}",
+                eodEntityToDelete.getMedia().getId(),
```

Comment:
Think you'll need to have another look at the logs and the params throughout.
Also what if the eod links to objects other than media, like transcription_document, then the current logging will NPE?